### ISSUE TITLE
vscode support: onview activation event doesn't work for a custom view in a custom view container

### ISSUE THREAD

**Issue by jcortell68** [eclipse-theia/theia#13157]
### Bug Description:
When a vscode extensions contributes a custom view container (and a custom view that is in that container), and it declares an activation event for the view being opened (`onview:myviewid`), the plugin doesn't get activated when opening the view and the view shows no content.

One workaround is to declare `*` as an activation event but that is obviously not ideal. We don't want the plugin activated if the user isn't going to open the view.

Also, for whatever reason, the view is properly activated if the app launches with the view already opened. I.e., if the user restarts/refreshes the app with the view already open, then the plugin is activated and everything works as expected. 

### Steps to Reproduce:
```
git clone -b webview_panel_noneditor https://github.com/jcortell68/theiasandbox
cd theiasandbox
yarn
yarn start:electron
```

1. Bring up the command palette (Ctrl+Shift+p on Windows) and trigger the `View: Reset Workbench Layout` command.
2. Wait for the app to refresh
3. Open the custom view via `View > Open View... > My Webview`
4. Notice the view comes up but is empty. It should be saying hello to the world.
5. Notice that the stdout lacks the `console.log()` that the plugin writes from it's activation event (`Congratulations, your extension "myext" is now active...`). This proves that the plugin was not activated, despite having this in its package.json
```
"activationEvents": ["onView:myext_myView"],
```
6. Refresh the app (Ctrl+r on Windows).
7. Notice the custom view is now showing the expected content.

For further proof that the activation event isn't working, change the activation event so that the plugin is activate on launch unconditionally
```
"activationEvents": ["*"],
```
Now try to reproduce the problem following the steps above and you'll see the empty view issue doesn't happen.

### Additional Information

- Operating System: Windows
- Theia Version: 1.43.1

**Comment by jcortell68** [comment]
@martin-fleck-at reproduced the issue on his machine

**Comment by msujew** [comment]
Maybe related to https://github.com/eclipse-theia/theia/issues/13090.

**Comment by jcortell68** [comment]
I'd say almost certainly. Thanks.

**Comment by msujew** [comment]
I've encountered this issue when trying to test https://github.com/eclipse-theia/theia/pull/13091. While the effects are very similar to the other issue, they have independent problem sources. My findings during a short debugging session:

* [`activateByView`](https://github.com/eclipse-theia/theia/blob/0055c8cb129e2b5ff5747d2afe072a388307e0b8/packages/plugin-ext/src/hosted/browser/hosted-plugin.ts#L661-L663) isn't triggered at all due to this issue.
* Seems like the reason is that when trying to [prepare the views inside the contributed view container](https://github.com/eclipse-theia/theia/blob/0055c8cb129e2b5ff5747d2afe072a388307e0b8/packages/plugin-ext/src/main/browser/view/plugin-view-registry.ts#L636), the views [aren't actually registered yet](https://github.com/eclipse-theia/theia/blob/0055c8cb129e2b5ff5747d2afe072a388307e0b8/packages/plugin-ext/src/main/browser/view/plugin-view-registry.ts#L351-L370). When stepping through the code with a debugger, you can see that the views are registered after the preparation step.

**Comment by jcortell68** [comment]
> I've encountered this issue when trying to test #13091. While the effects are very similar to the other issue, they have independent problem sources.

Oh wow. Very surprising indeed. Thanks for attempting to verify my issue while fixing the other one. There's no shortage of work for you Theia folks in 2024 ;-)

**Comment by msujew** [comment]
@jcortell68 It does support it see https://github.com/eclipse-theia/theia/pull/12167. The event generator for `onView` just wasn't implemented correctly, which has been fixed by https://github.com/eclipse-theia/theia/pull/13091.

**Comment by jcortell68** [comment]
Yep. I get that the `onView` support was broken. I just honestly didn't know if the 1.74.0 auto feature was in Theia or not. Of course, when I ran into my bug (described in this ticket), I didn't explicitly declare the activationEvent and thought that was what was causing the broken behavior. I was wrong, of course. But I just realized I didn't know if the auto-behavior was in Theia. Glad to know it is! Thanks
